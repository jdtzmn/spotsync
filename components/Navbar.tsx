import * as React from 'react'
import css from '../styles/components/navbar.scss';

class Navbar extends React.Component {
  state = {
    hasScrolled: false
  }

  constructor (props) {
    super(props)

    this.handleScroll = this.handleScroll.bind(this)
  }

  render () {
    const { hasScrolled } = this.state

    return (
      <div className={css.navbar}>
        <div className={hasScrolled ? css.fixed : css.default}>
          <h1 className={css.navbarTitle}>Spotsync.</h1>
        </div>
      </div>
    )
  }

  handleScroll () {
    const hasScrolled = window.scrollY > 0
    this.setState({
      hasScrolled
    })
  }

  componentDidMount () {
    document.addEventListener('scroll', this.handleScroll)
    this.handleScroll()
  }

  componentWillUnmount () {
    document.removeEventListener('scroll', this.handleScroll)
  }
}

export default Navbar
