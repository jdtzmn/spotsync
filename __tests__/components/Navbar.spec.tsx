import * as React from 'react'
import { shallow } from 'enzyme'
import Navbar from 'components/Navbar'

describe('Code', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Navbar />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
