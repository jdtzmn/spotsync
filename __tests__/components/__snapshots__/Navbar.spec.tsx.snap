// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code should render correctly 1`] = `
<div>
  <div>
    <h1>
      Spotsync.
    </h1>
  </div>
</div>
`;
